### PR TITLE
Allow Jinja2 templating to exist in separate files

### LINKED ISSUE #426: Optional flag for generated PinFile to be output to a file
Per discussions with @jaypoulz, it was brought up that generating templates was a great feature. One thing it could possibly use is output of the templated PinFile, with topology, including the optional layout and hooks data included.

The concept here might be to take the uhash value and write out a `pinfiles/Pinfile-<uhash>.generated` file. If the `--output-pinfile` flag is passed. An optional value could be passed to record the file elsewhere. Relative paths would be relative to the execution directory.

### LINKED ISSUE #428: Processing Jinja2 from a separate topology or layout file fails
Per @jaypoulz, LinchPin should allow for processing topology and layout files using Jinja2 in addition to a single PinFile.

Essentially, inputs might look like this: 

```
$ cat PinFile.jenkins
jenkins-slave:
  topology: jenkins-slave.yml
  {% if hooks %}
  hooks:
    {% if hooks.postUp %}
    post-up:
      {% if hooks.postUp.connectToMaster %}
      name: cinch
      type: shell
      actions:
        - cinch inventories/jenkins-slave.inventory
      {% endif %}
    {% endif %}
  {% endif %}

$ cat topologies/jenkins-slave.yml 
---
topology_name: jenkins-slave
resource_groups:
- resource_group_name: beaker-slaves
  resource_group_type: beaker
  resource_definitions:
  - role: bkr_server
    job_group: multiarch-qe
    whiteboard: {{ arch }} slave for multiarch testing
    recipesets:
    - distro: RHEL-7.4
      arch: {{ arch }}
      variant: Server
      count: 1
```

Generating PinFile data from this would look like so:

```
linchpin -vp PinFile.jenkins --template-data '{"arch": "x86_64", "hooks": { "postUp": { "connectToMaster": "true" }}}' up
 
--- PINFILE OUTPUT---
{
    "jenkins-slave": {
        "hooks": {
            "post-up": {
                "actions": [
                    "cinch inventories/jenkins-slave.inventory"
                ],
                "name": "cinch",
                "type": "shell"
            }
        },
        "topology": {
            "resource_groups": [
                {
                    "resource_definitions": [
                        {
                            "job_group": "multiarch-qe",
                            "recipesets": [
                                {
                                    "arch": "x86_64",
                                    "count": 1,
                                    "distro": "RHEL-7.4",
                                    "variant": "Server"
                                }
                            ],
                            "role": "bkr_server",
                            "whiteboard": "x86_64 slave for multiarch testing"
                        }
                    ],
                    "resource_group_name": "beaker-slaves",
                    "resource_group_type": "beaker"
                }
            ],
            "topology_name": "jenkins-slave"
        }
    }
}
```

### PR DESCRIPTION
**Pull request by herlo**
@jaypoulz pointed out that while Jinja2 template processing was working in a single PinFile, it failed when including the topology or layout files. He suggested LinchPin process those files as well. This PR addresses processing those files as well. Each file gets processed independently using the template-data passed.

Additionally, this output can be written out to a file during to an action 'up' task.

Fixes #428 
Fixes #426

### CODE REVIEW

File: linchpin/cli/__init__.py [hunk 1]
```diff
@@ -245,31 +249,30 @@ def lp_down(self, pinfile, targets='all'):
...
 
-    def find_topology(self, topology):
+    def find_include(self, filename, ftype='topology'):
         """
-        Find the topology to be acted upon. This could be pulled from a
-        registry.
+        Find the included file to be acted upon.
 
         :param topology:
             name of topology from PinFile to be loaded
```
Comment (p3ck) [inline linchpin/cli/__init__.py]: docs need updating.

File: linchpin/cli/__init__.py [hunk 2]
```diff
@@ -278,41 +281,27 @@ def _build(self, pf):
...
                 topology_data = pf[target]['topology']
 
             provision_data[target]['topology'] = topology_data
 
-
             layout_data = None
 
             if 'layout' in pf[target]:
                 if not isinstance(pf[target]['layout'], dict):
-                    self.set_evar('layout_file',
```
Comment (p3ck) [inline linchpin/cli/__init__.py]: set_evar('layout_file'..  is not longer needed?

Comment (herlo) [reply linchpin/cli/__init__.py]: Nope. layou_file is now layout_data. Essentially, we could set the evar, but it's not used in the playbooks anymore.

Comment (p3ck) [reply linchpin/cli/__init__.py]: ok - +1 then once you fix the incorrect doc string in my previous comment

File: linchpin/cli/__init__.py (hunk 1 above)
Comment (herlo) [reply linchpin/cli/__init__.py]: This has been updated in the code, and will be submitted in the next PR.

**Pull request merged**